### PR TITLE
PushAuthentication: Swift Update(s)

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
As part of a Notifications refactor (merging a couple helpers!), we're updating few things around. Starting off with PushAuthenticationManager:

- Switched **if** statements over to guard
- Corrected method visibility
- Nuked NSObject inheritance
- Minor touches here and there

Ref #6755

### To test:
No logic has been updated in this PR (other than if getting replaced by guard). In order to test, please, verify that the unit tests don't break.

Needs review: @frosty 
Sir, may i bug you with yet another review?

Thanks!!

### CODE REVIEW

File: WordPress/Classes/Utility/PushAuthenticationManager.swift [hunk 1]
```diff
@@ -8,21 +8,25 @@ import WordPressShared
...
-    // MARK: - Public Methods
-    //
-    open var alertControllerProxy = UIAlertControllerProxy()
-    open let pushAuthenticationService: PushAuthenticationService
+class PushAuthenticationManager {
 
-    override convenience init() {
+    /// AlertController Proxy
+    ///
+    var alertControllerProxy = UIAlertControllerProxy()
```
Comment (frosty) [inline WordPress/Classes/Utility/PushAuthenticationManager.swift]: I guess this is a `var` just for unit tests?

File: WordPress/Classes/Utility/PushAuthenticationManager.swift [hunk 2]
```diff
@@ -103,20 +103,19 @@ import WordPressShared
     ///
     /// - Parameter userInfo: Is the Notification's payload.
     ///
-    fileprivate func isNotificationExpired(_ userInfo: NSDictionary?) -> Bool {
-        let rawExpiration = userInfo?["expires"] as? TimeInterval
-        if rawExpiration == nil {
+    func isNotificationExpired(_ userInfo: NSDictionary?) -> Bool {
```
Comment (frosty) [inline WordPress/Classes/Utility/PushAuthenticationManager.swift]: _Super_ nitpicky, but above we have `isPushAuthenticationNotification` and here we have `isNotificationExpired`... can we make them consistent and perhaps remove `Push` from the first one? (the class has Push in the name anyway)

File: WordPress/Classes/Utility/PushAuthenticationManager.swift [hunk 3]
```diff
@@ -135,24 +134,30 @@ import WordPressShared
...
+// MARK: - Private Internal Constants
+//
+private extension PushAuthenticationManager {
 
-    // MARK: - Private Internal Constants
-    fileprivate let initialRetryCount                   = 0
-    fileprivate let maximumRetryCount                   = 3
-    fileprivate let minimumRemainingExpirationTime      = TimeInterval(5)
-    fileprivate let pushAuthenticationNoteType          = "push_auth"
+    struct Settings {
```
Comment (frosty) [inline WordPress/Classes/Utility/PushAuthenticationManager.swift]: Might be best as an `enum`, so it can't be inited.

**Comment by jleandroperez** [comment]
All comments addressed. RE: `var alertProxy...`, yes, i'm afraid that's this way for unit testing purposes.

@frosty thanks for the review sir!!

**Pull request merged**